common/filetoolkitwithundo.cpp

```diff
@@ -180,15 +123,33 @@ bool FileToolkitWithUndo::overlayMount(const std::string &lower,
     std::string fstype = "overlay";
     unsigned long flags = 0;
 
-    if (!createDirectory(lower)
-        || !createDirectory(upper)
-        || !createDirectory(work)
-        || !createDirectory(dst))
+    if (!m_create.createDirectory(lower)
+        || !m_create.createDirectory(upper)
+        || !m_create.createDirectory(work)
+        || !m_create.createDirectory(dst))
     {
+        m_create.rollBack();
         log_error() << "Failed to create lower/upper/work directory for overlayMount. lowerdir=" <<
                        lower << ", upperdir=" << upper << ", workdir=" << work;
         return false;
     }
+    m_create.clear();
```

Comment:
I don't really like this solution at all. Wouldn't it make more sense if one could have a local FileToolKitWithUndo instance that also has a cleanup handler API?

```
FileToolkKitWithUndo ft;
if (!ft.createDirectory(lower) || ...) {
   // Something went wrong
} else 
    m_cleanupHandlers.emplace_back(ft);
}
```